### ISSUE TITLE
Is it possible to know if a Batch has pending jobs?

### ISSUE THREAD

**Issue by deborasetton** [mperham/sidekiq#3719]
Hi,

Is it possible to know if a Batch has any pending jobs?

I mean pending as in "the job has not run yet, or the job is in the Retry queue, and will still run until retries are exhausted".

I thought `Sidekiq::Batch::Status#pending` could be used for this, but if one of the batch's jobs has failed permanently (i.e., has moved to the Dead queue and will not be retried anymore), this job is still marked as pending.

To put it differently, I would like to know if a Batch has finished, successfully or not. The `success` callback cannot be used for this (because some jobs might have failed), and the `complete` callback cannot be used because it will fire even if some jobs are still in the Retry queue.

**Comment by mperham** [comment]
There is no way to know that today.  Several people have raised this issue; it's a known shortcoming.

**Comment by deborasetton** [comment]
Thanks for the super quick reply! I missed the other issues mentioning this, sorry for this being a duplicate. 

I would suggest adding this observation to the Sidekiq Batches docs, it seems very relevant.
